spec/javascripts/helpers/markdown_editor_spec.js

```diff
@@ -220,6 +220,38 @@ describe("Diaspora.MarkdownEditor", function() {
     });
   });
 
+  describe("isPreviewMode", function() {
+    beforeEach(function() {
+      this.target = new Diaspora.MarkdownEditor(this.$el, {onPreview: $.noop, onPostPreview: $.noop()});
+    });
+
+    it("return false if editor is not visible yet", function() {
+      this.target.instance = undefined;
+      expect(this.target.isPreviewMode()).toBe(false);
+    });
+
+    it("returns true if editor is in preview mode", function() {
+      this.target.showPreview();
+      expect(this.target.isPreviewMode()).toBe(true);
+    });
+  });
+
+  describe("userInputNotEmpty", function() {
+    beforeEach(function() {
+      this.target = new Diaspora.MarkdownEditor(this.$el, {onPreview: $.noop, onPostPreview: $.noop()});
+    });
+
+    it("return false if editor is not visible yet", function() {
+      this.target.instance = undefined;
+      expect(this.target.userInputNotEmpty()).toBe(false);
+    });
+
```

Comment:
As written above, I think this method should be called `userInputEmpty` and behave accordingly. Afterwards please add a test `returns true if the editor is empty`.